src/init/features/functions/index.js

```diff
@@ -22,35 +24,69 @@ module.exports = function(setup, config) {
 
   setup.functions = {};
   var projectId = _.get(setup, "rcfile.projects.default");
+  var requiredApis = ["cloudfunctions.googleapis.com", "runtimeconfig.googleapis.com"];
   var enableApis;
   if (projectId) {
     enableApis = requireAccess({ project: projectId }, [scopes.CLOUD_PLATFORM]).then(function() {
-      enableApi(projectId, "cloudfunctions.googleapis.com");
-      enableApi(projectId, "runtimeconfig.googleapis.com");
+      return promiseAllSettled(requiredApis.map((api) => enableApi(projectId, api)));
     });
   } else {
-    enableApis = Promise.resolve();
+    enableApis = Promise.all([]);
   }
-  return enableApis.then(function() {
-    return prompt(setup.functions, [
-      {
-        type: "list",
-        name: "language",
-        message: "What language would you like to use to write Cloud Functions?",
-        default: "javascript",
-        choices: [
+  return enableApis
+    .then(function(results) {
+      var disabledApis = [];
+      results
+        .filter((result) => result.state === "rejected" && result.reason.status === 403)
+        .forEach((value, index) => disabledApis.push(requiredApis[index]));
+      if (disabledApis.length > 0) {
+        var options = {};
+        return prompt(options, [
           {
-            name: "JavaScript",
-            value: "javascript",
+            type: "list",
+            name: "continue",
+            message: `You don't have permission to enable the APIs : ${disabledApis.join(
+              ", "
+            )}.\nYou will not be able to deploy the functions while they are disabled. Do you want to proceed?`,
+            default: "yes",
+            choices: [
+              {
+                name: "Yes",
+                value: "yes",
+              },
+              {
+                name: "No",
+                value: "no",
+              },
+            ],
           },
-          {
-            name: "TypeScript",
-            value: "typescript",
-          },
-        ],
-      },
-    ]).then(function() {
-      return require("./" + setup.functions.language)(setup, config);
+        ]).then(function() {
+          if (options.continue === "no") {
+            return Promise.reject(new FirebaseError("Initialization aborted."));
+          }
+        });
+      }
```

Comment:
Actually that's what I first though of, but then I thought that it should be up to the user to decide whether to abort or proceed. But as you said, it could be a bit confusing to the user.